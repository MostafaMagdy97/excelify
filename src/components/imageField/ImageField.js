import React, { Component } from 'react';

class ImageField extends Component {
  onChange = () => {};

  setProps = (onChange, callbackParams) => {
    this.onChange = onChange;
    this.callbackParams = callbackParams;
  };

  callbackParams;

  uploadPhoto = () => {
    const file = this.photoInput.files[0];
    const fileName = this.photoInput.files[0].name;
    this.photoInput.value = '';
    const fileReader = new FileReader();

    fileReader.onloadend = () => {
      this.onChange(fileReader.result, fileName, this.callbackParams);
    };

    if (file) {
      fileReader.readAsDataURL(file);
    }
  };

  click = () => {
    this.photoInput.click();
  };

  render() {
    return (
      <input
        type="file"
        accept=".png,.jpg,.jpeg"
        onChange={this.uploadPhoto}
        ref={photoInput => {
          this.photoInput = photoInput;
        }}
        style={{ display: 'none' }}
      />
    );
  }
}

export default ImageField;
