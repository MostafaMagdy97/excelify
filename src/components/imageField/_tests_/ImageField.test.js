import React from 'react';
import { configure, render } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-enzyme';
import 'jest-styled-components';

import ImageField from '../ImageField';

configure({ adapter: new Adapter() });

describe('ImageField', () => {
  it('should render', () => {
    const tree = render(<ImageField />);
    expect(tree).toMatchSnapshot();
  });
});
