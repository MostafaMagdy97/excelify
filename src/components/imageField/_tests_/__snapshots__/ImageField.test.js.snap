// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageField should render 1`] = `
<input
  accept=".png,.jpg,.jpeg"
  style="display:none"
  type="file"
>
  
</input>
`;
